src/components/ha-form/ha-form-dictionary.ts

```diff
@@ -0,0 +1,127 @@
+import "../ha-icon-button";
+import "@polymer/paper-input/paper-input";
+import {
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+  css,
+  CSSResultArray,
+  PropertyValues,
+  internalProperty,
+} from "lit-element";
+import { fireEvent } from "../../common/dom/fire_event";
+import type {
+  HaFormElement,
+  HaFormDictionaryData,
+  HaFormDictionarySchema,
+  HaFormSchema,
+  HaFormBooleanSchema,
+} from "./ha-form";
+
+@customElement("ha-form-dictionary")
+export class HaFormDictionary extends LitElement implements HaFormElement {
+  @property() public schema!: HaFormDictionarySchema;
+
+  @property() public data!: HaFormDictionaryData;
+
+  @property() public label!: string;
+
+  @property() public suffix!: string;
+
+  @property() public computeError?: (schema: HaFormSchema, error) => string;
+
+  @property() public computeLabel?: (nesting: string, param?: string) => string;
+
+  @property() public computeSuffix?: (schema: HaFormSchema) => string;
+
+  @property() public nesting = "";
+
+  @internalProperty() private _showParams = true;
+
+  private _saveData = this.data || ({} as HaFormDictionaryData);
+
+  protected render(): TemplateResult {
+    const boolSchema = { type: "boolean" } as HaFormBooleanSchema;
+    return html`
+      ${this._showParams
+        ? html`
+            <hr></hr>
+            <h4>${this._computeLabel("title")}</h4>
+            ${this._computeLabel("description")}
+          `
+        : ""}
+      ${this.schema.optional
+        ? html`
+            <ha-form-boolean
+              .schema=${boolSchema}
+              .data=${this._showParams}
+              label=${this._computeLabel("optional") ||
+              `Configure ${this._computeLabel("title")}`}
+              @value-changed=${this._showParamsChanged}
+            ></ha-form-boolean>
+          `
+        : ""}
```

Comment:
This should not be a `ha-form`, but just a checkbox

Reply:
i can change, but why not use ha-form which is already styled accordingly? it has a separate callback function, and can be treated as two separate forms